### PR TITLE
Fix Incorrect Lengths SDEs on Computed Element with LengthKind Implicit

### PR DESCRIPTION
**Pull request by olabusayoT**
- add condition that checks for IVC before doing the check
- add tests

DAFFODIL-2806

### CODE REVIEW

File: daffodil-core/src/main/scala/org/apache/daffodil/core/dsom/ElementBase.scala [hunk 1]
```diff
@@ -1007,13 +1007,19 @@ trait ElementBase
...
         )
-        val res = (hasLength, hasMinLength, hasMaxLength, lengthKind) match {
-          case (true, false, false, _) => (r.lengthValue, r.lengthValue)
-          case (true, _, _, _) =>
+        val res = (
+          hasLength,
+          hasMinLength,
+          hasMaxLength,
+          lengthKind,
+          inputValueCalcOption.isDefined
```
Comment (stevedlawrence) [inline daffodil-core/src/main/scala/org/apache/daffodil/core/dsom/ElementBase.scala:1015]: It's better to use isRepresented to determine if something is IVC or not.

**Review by stevedlawrence — APPROVED** [review]
+1, minor cleanup to use isRepresented instead of explicitly checking the IVC property

**Review by mbeckerle — APPROVED** [review]
+1

**Review by jadams-tresys — APPROVED** [review]
+1

**Pull request merged**